src/game/mod.rs

```diff
@@ -176,8 +178,10 @@ impl GameManager {
             sessions: DashMap::new(),
         }
     }
-
-    pub fn create_game(&self, game: impl FnOnce(GameId) -> Box<dyn GameAdapter>) -> Result<GameId> {
+    pub fn create_game(
+        &mut self,
+        game: impl FnOnce(GameId) -> Box<dyn GameAdapter>,
+    ) -> Result<GameId> {
```

Comment:
Please do not mutably borrow self here — it is not necessary and it stops the project building